### ISSUE TITLE
[Feature request] run application

### ISSUE THREAD

**Issue by ToBinio** [HuakunShen/applications-rs#5]
It would be great to have a cross-platform way to start applications.

The current docs say that the `app_path_exe` is the path to the executable which at least for Linux isn't the case since it "just" a start command that can even contain parameter placeholders - https://specifications.freedesktop.org/desktop-entry-spec/latest/exec-variables.html

so it would be nice to have some .run() function that consumes params and runs the app

notes:
I would be happy to try my best to help implement this for Linux 
the type of `app_path_exe` should probably be changed since it isn't a path on linux

Thanks for the library!

**Comment by HuakunShen** [comment]
Yeah the app_path_exe field is intended to be the executable path, but the parsed value from `.desktop` file don't work all the time.
It's simple on Mac, all `.app` can be started with `open`.

 I would be grateful if you could offer some help with Linux. Do you have any suggestions for changes?

**Comment by ToBinio** [comment]
I believe it should be fine to simply change the `app_path_exe` to a `String`

---

and in terms of handling the `exec` command - i would image an app running api similar to this:

```rust
impl App {
    fn run() -> AppRunner {}
}

pub struct AppRunner {}

impl AppRunner {
    
    /// sets a path as an argument for the app 
    /// %f | %F | %u | %U on linux
    fn path(self, path: PathBuf) -> AppRunner {}

    /// sets multiple paths as an argument for the app 
    /// %F | %U on linux
    fn paths(self, paths: Vec<PathBuf>) -> AppRunner {}

    /// sets an url as an argument for the app 
    /// %u | %U on linux
    fn url(self, url: String) -> AppRunner {}

    /// sets multiple urls as an argument for the app 
    /// %U on linux
    fn urls(self, urls: Vec<String>) -> AppRunner {}

    /// adds an argument for the app 
    fn arg(self, arg: String) -> AppRunner {}
    
    /// adds multiple arguments for the app
    fn args(self, args: Vec<String>) -> AppRunner {}

    /// starts the app
    fn spawn() -> Child {}

    /// creates command that can be used to start the app 
    fn command() -> Command {}
}
```
note: 
function definition (name, types, ...) are just for example

example usage:

```rust
some_app.run().path(path_to_some_folder).arg("--force").spawn()
```

this should also be fine for mac & windows
e.g for mac all the path & URL functions could just to the same as arg & do some type validation

**Comment by Gibbz** [comment]
any progress on this? im looking for something that works on linux

**Comment by HuakunShen** [comment]
I was still busy working on another project. I may have time to work on this next week.

**Comment by HuakunShen** [comment]
I am back to this project.
